tests/test_permissions.py

```diff
@@ -0,0 +1,186 @@
+    _check_permission_denied(engine, "level_a1", "gebieden_buurten")
+    _check_permission_denied(engine, "level_b1", "gebieden_buurten")
+    _check_permission_denied(engine, "level_c1", "gebieden_buurten")
+    _check_permission_denied(engine, "level_a2", "gebieden_bouwblokken")
+    _check_permission_denied(engine, "level_b2", "gebieden_bouwblokken")
+    _check_permission_denied(engine, "level_c2", "gebieden_bouwblokken")
+    _check_permission_denied(engine, "level_a2", "gebieden_buurten")
+    _check_permission_denied(engine, "level_b2", "gebieden_buurten")
+    _check_permission_denied(engine, "level_c2", "gebieden_buurten")
+
+    # Apply the permissions from Schema and Profiles.
+    apply_schema_and_profile_permissions(engine, ams_schema, profiles, "level_a1", "LEVEL/A1")
+    apply_schema_and_profile_permissions(engine, ams_schema, profiles, "level_b1", "LEVEL/B1")
+    apply_schema_and_profile_permissions(engine, ams_schema, profiles, "level_c1", "LEVEL/C1")
+    apply_schema_and_profile_permissions(engine, ams_schema, profiles, "level_a2", "LEVEL/A2")
+    apply_schema_and_profile_permissions(engine, ams_schema, profiles, "level_b2", "LEVEL/B2")
+    apply_schema_and_profile_permissions(engine, ams_schema, profiles, "level_c2", "LEVEL/C2")
+    # Check if the read priviliges are correct
+    _check_permission_denied(engine, "level_a1", "gebieden_bouwblokken")
+    _check_permission_granted(engine, "level_b1", "gebieden_bouwblokken", "id, eind_geldigheid")
+    _check_permission_denied(engine, "level_b1", "gebieden_bouwblokken", "begin_geldigheid")
+    _check_permission_denied(engine, "level_c1", "gebieden_bouwblokken", "id, eind_geldigheid")
+    _check_permission_granted(engine, "level_c1", "gebieden_bouwblokken", "begin_geldigheid")
+    _check_permission_granted(engine, "level_a1", "gebieden_buurten")
+    _check_permission_denied(engine, "level_b1", "gebieden_buurten")
+    _check_permission_denied(engine, "level_c1", "gebieden_buurten")
+    _check_permission_denied(engine, "level_a2", "gebieden_bouwblokken")
+    _check_permission_granted(engine, "level_b2", "gebieden_bouwblokken", "id, eind_geldigheid")
+    _check_permission_denied(engine, "level_b2", "gebieden_bouwblokken", "begin_geldigheid")
+    _check_permission_denied(engine, "level_c2", "gebieden_bouwblokken", "id, eind_geldigheid")
+    _check_permission_granted(engine, "level_c2", "gebieden_bouwblokken", "begin_geldigheid")
+    _check_permission_granted(engine, "level_a2", "gebieden_buurten")
+    _check_permission_denied(engine, "level_b2", "gebieden_buurten")
+    _check_permission_denied(engine, "level_c2", "gebieden_buurten")
+
+def _create_role(engine, role):
+    #  If role already exists just fail and ignore. This may happen if a previous pytest did not terminate correctly.
+    try:
+        engine.execute('CREATE ROLE "{}"'.format(role))
+    except ProgrammingError:
+        #  psycopg2.errors.DuplicateObject
```

Comment:
```suggestion
    except psycopg2.errors.DuplicateObject:
```

Reply:
Het vervelende was dat die psycopg2 error wordt afgevangen door die ProgrammingError, waardoor ik nu minder specifiek ben dan ik zou willen. Ik ga er nog even naar kijken of dit niet anders kan.

Reply:
```
try:
        engine.execute('CREATE ROLE "{}"'.format(role))
except ProgrammingError as e:
        if not isinstance(e.orig, DuplicateObject):
            raise
```